bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/connection/HubConnection.java

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nobohub.internal.connection;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.nobohub.internal.Helpers;
+import org.openhab.binding.nobohub.internal.NoboHubBindingConstants;
+import org.openhab.binding.nobohub.internal.NoboHubBridgeHandler;
+import org.openhab.binding.nobohub.internal.model.Hub;
+import org.openhab.binding.nobohub.internal.model.NoboCommunicationException;
+import org.openhab.binding.nobohub.internal.model.NoboDataException;
+import org.openhab.binding.nobohub.internal.model.OverrideMode;
+import org.openhab.binding.nobohub.internal.model.OverridePlan;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Connection to the Nobø Hub (Socket wrapper).
+ *
+ * @author Jørgen Austvik - Initial contribution
+ * @author Espen Fossen - Initial contribution
+ */
+@NonNullByDefault
+public class HubConnection {
+
+    private final Logger logger = LoggerFactory.getLogger(HubConnection.class);
+
+    private final InetAddress host;
+    private final NoboHubBridgeHandler hubHandler;
+    private final String serialNumber;
+
+    private @Nullable Socket hubConnection;
+    private @Nullable PrintWriter out;
+    private @Nullable BufferedReader in;
+
+    public HubConnection(String hostName, String serialNumber, NoboHubBridgeHandler hubHandler)
+            throws NoboCommunicationException {
+        try {
+            host = InetAddress.getByName(hostName);
+        } catch (IOException ioex) {
+            throw new NoboCommunicationException(String.format("Failed to resolve IP address of %s", hostName));
+        }
```

Comment:
It might be a good idea to avoid I/O in the constructor. You could store the hostname as instance variable and move the `getByName()` call to `connectSocket()`, where the result could still be cached after first lookup.

Reply:
Added a proposed solution, please check if this is what you was looking for,

Reply:
Looks good!